tests/bpf2c_tests/expected/bindmonitor_bpf2bpf_sys.c

```diff
@@ -220,92 +218,211 @@ BindMonitor_Caller(void* context)
+    *(uint32_t*)(uintptr_t)(r10 + OFFSET(-4)) = (uint32_t)r1;
+    // EBPF_OP_MOV64_IMM pc=27 dst=r1 src=r0 offset=0 imm=1
+#line 41 "sample/bindmonitor_bpf2bpf.c"
+    r1 = IMMEDIATE(1);
+    // EBPF_OP_STXW pc=28 dst=r10 src=r1 offset=-36 imm=0
+#line 41 "sample/bindmonitor_bpf2bpf.c"
+    *(uint32_t*)(uintptr_t)(r10 + OFFSET(-36)) = (uint32_t)r1;
+    // EBPF_OP_JA pc=29 dst=r0 src=r0 offset=13 imm=0
+#line 41 "sample/bindmonitor_bpf2bpf.c"
+    goto label_8;
+label_5:
+    // EBPF_OP_LDXDW pc=30 dst=r1 src=r10 offset=-32 imm=0
+#line 44 "sample/bindmonitor_bpf2bpf.c"
+    r1 = *(uint64_t*)(uintptr_t)(r10 + OFFSET(-32));
+    // EBPF_OP_JNE_IMM pc=31 dst=r1 src=r0 offset=6 imm=1
+#line 44 "sample/bindmonitor_bpf2bpf.c"
+    if (r1 != IMMEDIATE(1)) {
+#line 44 "sample/bindmonitor_bpf2bpf.c"
+        goto label_7;
+#line 44 "sample/bindmonitor_bpf2bpf.c"
+    }
+    // EBPF_OP_JA pc=32 dst=r0 src=r0 offset=0 imm=0
+#line 44 "sample/bindmonitor_bpf2bpf.c"
+    goto label_6;
+label_6:
+    // EBPF_OP_MOV64_IMM pc=33 dst=r1 src=r0 offset=0 imm=2
+#line 44 "sample/bindmonitor_bpf2bpf.c"
+    r1 = IMMEDIATE(2);
+    // EBPF_OP_STXW pc=34 dst=r10 src=r1 offset=-4 imm=0
+#line 46 "sample/bindmonitor_bpf2bpf.c"
+    *(uint32_t*)(uintptr_t)(r10 + OFFSET(-4)) = (uint32_t)r1;
+    // EBPF_OP_MOV64_IMM pc=35 dst=r1 src=r0 offset=0 imm=1
+#line 46 "sample/bindmonitor_bpf2bpf.c"
+    r1 = IMMEDIATE(1);
+    // EBPF_OP_STXW pc=36 dst=r10 src=r1 offset=-36 imm=0
+#line 46 "sample/bindmonitor_bpf2bpf.c"
+    *(uint32_t*)(uintptr_t)(r10 + OFFSET(-36)) = (uint32_t)r1;
+    // EBPF_OP_JA pc=37 dst=r0 src=r0 offset=5 imm=0
+#line 46 "sample/bindmonitor_bpf2bpf.c"
+    goto label_8;
+label_7:
+    // EBPF_OP_MOV64_IMM pc=38 dst=r1 src=r0 offset=0 imm=0
+#line 46 "sample/bindmonitor_bpf2bpf.c"
+    r1 = IMMEDIATE(0);
+    // EBPF_OP_STXW pc=39 dst=r10 src=r1 offset=-4 imm=0
+#line 48 "sample/bindmonitor_bpf2bpf.c"
+    *(uint32_t*)(uintptr_t)(r10 + OFFSET(-4)) = (uint32_t)r1;
+    // EBPF_OP_MOV64_IMM pc=40 dst=r1 src=r0 offset=0 imm=1
+#line 48 "sample/bindmonitor_bpf2bpf.c"
+    r1 = IMMEDIATE(1);
+    // EBPF_OP_STXW pc=41 dst=r10 src=r1 offset=-36 imm=0
+#line 48 "sample/bindmonitor_bpf2bpf.c"
+    *(uint32_t*)(uintptr_t)(r10 + OFFSET(-36)) = (uint32_t)r1;
+    // EBPF_OP_JA pc=42 dst=r0 src=r0 offset=0 imm=0
+#line 48 "sample/bindmonitor_bpf2bpf.c"
+    goto label_8;
+label_8:
+    // EBPF_OP_LDXW pc=43 dst=r0 src=r10 offset=-4 imm=0
+#line 49 "sample/bindmonitor_bpf2bpf.c"
+    r0 = *(uint32_t*)(uintptr_t)(r10 + OFFSET(-4));
+    // EBPF_OP_EXIT pc=44 dst=r0 src=r0 offset=0 imm=0
+#line 49 "sample/bindmonitor_bpf2bpf.c"
     return r0;
 #line 27 "sample/bindmonitor_bpf2bpf.c"
 }
 #pragma code_seg(pop)
 #line __LINE__ __FILE__
 
 static uint64_t
-BindMonitor_Callee(uint64_t r1, uint64_t r2, uint64_t r3, uint64_t r4, uint64_t r5, uint64_t r10)
+BindMonitor_Callee(uint64_t r1, uint64_t r2, uint64_t r3, uint64_t r4, uint64_t r5, uint64_t r10, void* context)
```

Comment:
it seems the callee also has `context` as a param in the generated code. Is this expected?

Reply:
And actually the caller is passing only `r1` to `r10` when calling `BindMonitor_Callee`

Reply:
> And actually the caller is passing only `r1` to `r10` when calling `BindMonitor_Callee`

No, line 250 passes context too.